main/src/library/IpAddr.flix

```diff
@@ -38,5 +40,27 @@ mod IpAddr {
         case _ => unreachable!()
     }
 
+    ///
+    /// Converts an `IpAddr` into an `Array[Int8, r]`
+    ///
+    /// an Ipv4 is 4 bytes and Ipv6 is 16 bytes
+    ///
+    pub def toByteArray(rc: Region[r], ip: IpAddr): Array[Int8, r] \ r =
+        match ip {
+            case IpAddr.IPv4(b1, b2, b3, b4) =>
+                Array#{b1, b2, b3, b4} @ rc
+            case IpAddr.IPv6(b1, b2, b3, b4, b5, b6, b7, b8, b9, b10, b11, b12, b13, b14, b15, b16) =>
+                Array#{b1, b2, b3, b4, b5, b6, b7, b8, b9, b10, b11, b12, b13, b14, b15, b16} @ rc
+        }
+
+    ///
+    /// Converts an `IpAddr` into an `InetAddress`
+    ///
+    pub def toInetAddress(ip: IpAddr): InetAddress \ {IO, Net} =
+        region r {
+            let bytes = IpAddr.toByteArray(r, ip);
+            InetAddress.getByAddress(bytes)
```

Comment:
From [the docs](https://docs.oracle.com/javase/8/docs/api/java/net/InetAddress.html#getByAddress-byte:A-):

> Throws:
    [UnknownHostException](https://docs.oracle.com/javase/8/docs/api/java/net/UnknownHostException.html) - if IP address is of illegal length

but I am not handling that since the type shouldn't represent ips of illegal lengths